components/textbox/textbox.tsx

```diff
@@ -66,257 +67,277 @@ type Props = {
-        if (textbox) {
-            textbox.focus();
-            Utils.placeCaretAtEnd(textbox);
-            setTimeout(() => {
-                Utils.scrollToCaret(textbox);
-            });
-
-            // reset character count warning
-            this.checkMessageLength(textbox.value);
-        }
-    }
+    useImperativeHandle(ref, () => ({
+        getInputBox: () => message.current?.getTextbox(),
 
-    blur = () => {
-        this.getInputBox()?.blur();
-    };
+        focus: () => {
+            const textbox = getInputBox();
+            if (textbox) {
+                textbox.focus();
+                Utils.placeCaretAtEnd(textbox);
+                setTimeout(() => {
+                    Utils.scrollToCaret(textbox);
+                });
 
-    render() {
-        let preview = null;
+                // reset character count warning
+                checkMessageLength(textbox.value);
+            }
+        },
+        blur: () => getInputBox()?.blur(),
+    }));
 
-        let textboxClassName = 'form-control custom-textarea';
-        let textWrapperClass = 'textarea-wrapper';
-        if (this.props.emojiEnabled) {
-            textboxClassName += ' custom-textarea--emoji-picker';
-        }
-        if (this.props.badConnection) {
-            textboxClassName += ' bad-connection';
-        }
-        if (this.props.preview) {
-            textboxClassName += ' custom-textarea--preview';
-            textWrapperClass += ' textarea-wrapper--preview';
-
-            preview = (
-                <div
-                    tabIndex={this.props.tabIndex || 0}
-                    ref={this.preview}
-                    className='form-control custom-textarea textbox-preview-area'
-                    onKeyPress={this.props.onKeyPress}
-                    onKeyDown={this.handleKeyDown}
-                    onBlur={this.handleBlur}
-                >
-                    <PostMarkdown
-                        isRHS={this.props.isRHS}
-                        message={this.props.value}
-                        mentionKeys={[]}
-                        channelId={this.props.channelId}
-                    />
-                </div>
-            );
+    useEffect(() => {
+        if (preview) {
+            previewDiv.current?.focus();
         }
+        updateSuggestions({autocompleteGroups, channelId, currentTeamId, currentUserId, priorityProfiles, useChannelMentions, rootId, value});
+    }, [preview, previewDiv, updateSuggestions, autocompleteGroups, channelId, currentTeamId, currentUserId, priorityProfiles, useChannelMentions, rootId, value]);
+
+    checkMessageLength(value);
+    let previewText = null;
 
-        return (
+    let textboxClassName = 'form-control custom-textarea';
+    let textWrapperClass = 'textarea-wrapper';
+    if (emojiEnabled) {
+        textboxClassName += ' custom-textarea--emoji-picker';
+    }
+    if (badConnection) {
+        textboxClassName += ' bad-connection';
+    }
+    if (preview) {
```

Comment:
Can probably break this to another component to reduce lines here and clean it up